variables.tf

```diff
@@ -0,0 +1,140 @@
+variable "iam_role_name" {
+  description = "Name of IAM Role to associate to the Backup Plan"
+  type        = string
+  default     = null
+}
+
+# Rules
+variable "rules" {
+  description = "A list of rules mapping rule configurations for a backup plan"
+  type        = any
+  default     = []
+}
+
+# Selection
+variable "selection_name" {
+  description = "The display name of a resource selection document"
+  type        = string
+  default     = null
+}
+
+variable "selection_resources" {
+  description = "An array of strings that either contain Amazon Resource Names (ARNs) or match patterns of resources to assign to a backup plan"
+  type        = list
```

Comment:
list(string) ? Based on description it is an array of strings? 

Btw, I would rename description to state that it is a list of strings